src/golang/cmd/server/handler/list_artifact_results.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/aqueducthq/aqueduct/lib/collections/artifact_result"
+	"github.com/aqueducthq/aqueduct/lib/collections/workflow_dag"
+	aq_context "github.com/aqueducthq/aqueduct/lib/context"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/storage"
+	"github.com/aqueducthq/aqueduct/lib/workflow/artifact"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+)
+
+// Route: /artifact/{artifactId}/results
+// Method: GET
+// Params: None
+// Request:
+//	Headers:
+//		`api-key`: user's API Key
+// Response:
+//	Body:
+//		serialized `listArtifactResultsResponse`
+
+type listArtifactResultsResponse struct {
+	Results []artifact.RawResultResponse `json:"results"`
+}
+
+type listArtifactResultsArgs struct {
+	*aq_context.AqContext
+	ArtifactId uuid.UUID
+}
+
+type ListArtifactResultsHandler struct {
+	GetHandler
+
+	Database             database.Database
+	ArtifactReader       db_artifact.Reader
+	ArtifactResultReader artifact_result.Reader
+	WorkflowDagReader    workflow_dag.Reader
+}
+
+func (*ListArtifactResultsHandler) Name() string {
+	return "ListArtifactResults"
+}
+
+func (*ListArtifactResultsHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	aqContext, statusCode, err := aq_context.ParseAqContext(r.Context())
+	if err != nil {
+		return nil, statusCode, err
+	}
+
+	artfIdStr := chi.URLParam(r, routes.ArtifactIdUrlParam)
+	artfId, err := uuid.Parse(artfIdStr)
+	if err != nil {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "Malformed artifact ID.")
+	}
+
+	return &listArtifactResultsArgs{
+		AqContext:  aqContext,
+		ArtifactId: artfId,
+	}, http.StatusOK, nil
+}
+
+func (h *ListArtifactResultsHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*listArtifactResultsArgs)
+	artfId := args.ArtifactId
+
+	emptyResponse := listArtifactResultsResponse{}
+
+	results, err := h.ArtifactResultReader.GetArtifactResultsByArtifactId(ctx, artfId, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve artifact results.")
+	}
+
+	artf, err := h.ArtifactReader.GetArtifact(ctx, artfId, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve artifact.")
+	}
+
+	if len(results) == 0 {
+		return emptyResponse, http.StatusOK, nil
+	}
+
+	resultIds := make([]uuid.UUID, 0, len(results))
+	for _, result := range results {
+		resultIds = append(resultIds, result.Id)
+	}
+
+	dbDagsByResultIds, err := h.WorkflowDagReader.GetWorkflowDagsMapByArtifactResultIds(ctx, resultIds, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve workflow dags.")
+	}
+
+	// maps from db dag Ids
+	dbDagsMapByDagIds := make(map[uuid.UUID]workflow_dag.DBWorkflowDag, len(dbDagsByResultIds))
```

Comment:
nit on naming: I don't think you need the `Map` phrase since it's implicit in the "<something>By<something>" pattern, and I think it should use singular terminology since its one dag id per one dag, so this would be `dbDagByDagID`. I've been using that convention throughout the codebase so it might be good to settle on a consistent pattern?

Same for all the other maps here.